### ISSUE TITLE
[rllib][tune] How do you correctly use tensorflow variables in Population Based Training (PBT) reset_config?

### ISSUE THREAD

**Issue by internetcoffeephone** [ray-project/ray#8675]
### What is your question?

I want to run PBT on my multi-agent PPO experiment. The [provided example](https://rllib.readthedocs.io/en/latest/tune/generated_guides/plot_tune-advanced-tutorial.html#trainable-api-with-population-based-training) directly updates the learning rate like so: `param_group["lr"] = new_config["lr"]`.

With tensorflow, this is not directly possible. One of the parameters I'd like to update is the `entropy_coeff`, for which [EntropyCoeffSchedule](https://github.com/ray-project/ray/blob/releases/0.8.5/rllib/policy/tf_policy.py#L694) is required by ppo_tf_policy. This, however, is a tensorflow variable which can't be simply assigned to, instead we have to load it into the existing variable. When I load the variable like this:
```
def reset_config(self, new_config):
    for policy in self.optimizer.policies.values():
        policy.entropy_coeff.load(new_config["entropy_coeff"], session=self.optimizer.sess)
        policy.config["entropy_coeff"] = new_config["entropy_coeff"]
    self.config = new_config
```
the results will show that the value is being perturbed, but `entropy_coeff` is reported as the old value:
```
experiment_tag: 0_entropy_coeff=3.1357e-05@perturbed[entropy_coeff=0.00046194]
info:
  grad_time_ms: 761.994
  learner:
    agent-0:
      cur_kl_coeff: 0.012500000186264515
      cur_lr: 0.0009998156456276774
      entropy: 2.196960687637329
      entropy_coeff: 3.1357252737507224e-05
```

Is this simply a reporting issue, or is the actual value not changed?

*Ray version and other system information (Python version, TensorFlow version, OS):*
ray[rllib]==0.8.5

**Comment by internetcoffeephone** [comment]
I solved this problem, and have two things to report:

- The `experiment_tag` reporting is "ahead" by one experiment, so `entropy_coeff` will show not show the perturbed value the first time it shows up.
- The EntropyCoeffSchedule object was overriding the loaded variable with a value retrieved by `policy.entropy_coeff_schedule.value(_)`. Hence, the way to change this without having to change either ray or non-tuning code is the following:

```
def reset_config(self, new_config):
    for policy in self.optimizer.policies.values():
        policy.entropy_coeff_schedule.value = lambda _: new_config["entropy_coeff"]
        policy.config["entropy_coeff"] = new_config["entropy_coeff"]
    self.config = new_config
    return True
```
